### ISSUE TITLE
Detailed architecture of the SAM (Segment anything model)

### ISSUE THREAD

**Issue by jetsonwork** [facebookresearch/segment-anything#684]
Hello everyone,

I am trying to draw the architecture of SAM (not the one available in the original paper), including all the details. Could you please guide me on how to proceed? if someone has already done this, I would appreciate it if you could share it with me.

**Comment by heyoeyo** [comment]
For the image encoder specifically, the SAM model uses the 'plain' architecture described in the paper:
"[Exploring Plain Vision Transformer Backbones for Object Detection](https://arxiv.org/abs/2203.16527)"

However, if you're looking for 'all the details', then the code is definitely the best place to be looking. All of the model code is under the [segment_anything > modeling](https://github.com/facebookresearch/segment-anything/tree/main/segment_anything/modeling) folder, and it's very well organized/simple compared to most other model implementations that I've seen.

**Comment by jetsonwork** [comment]
Thanks. I have studied some blogs and they have mentioned the encoder architecture is based on MAE ViT as below. Could you please confirm it?

<img width="334" alt="image" src="https://github.com/facebookresearch/segment-anything/assets/125987810/294475b8-c887-4710-9685-0b589c1edee2">

**Comment by heyoeyo** [comment]
Yes the paper mentions they started with a model trained as a MAE, though the actual model in SAM isn't an autoencoder, I assume they just removed the decoder part.
